### PR TITLE
Display a guild's DAOs on their page and support more than one

### LINKED ISSUE #1142: [MyMeta] Add DAO info to guild pages
Right now, there are no links to guild's DAO even if they have one.

I think we should add a fiield for a link to "your DAO", "your multisig" &/or "your treasury" to the guild application form. 
Every guild that has a DAO on DAOHaus or a multisig, should also have a link on their profile connected to it.

### PR DESCRIPTION
**Pull request by alalonde**
## Overview

There are a few outstanding issues with our DAO / guild / player relationships, with the main end-goal being having a new set of tables (dao and dao_players) that we can use as a cache to avoid the very expensive daohaus queries we're currently making.  My plan for this is:

1. Create those two new tables, including a guild_id in the dao table so that guilds can associate one or more daos / multisigs / treasury addresses. Also add the relevant fields currently being returned from daohaus 
2. Refactor the current guild / moloch_address persistence code to instead persist 1 or more daos. This includes updating the guild create / edit form 
3. Add any of a guild's inputted DAO data to the guild page
4. Remove the current daohaus remote schema relationship for a player in favor of a more generic action to be called to update the cached data in the new dao tables
5. Implement the cache update using the current IDX player caching logic as a model

This PR tackles 1 - 3, 4/5 will be addressed in later PRs.

Closes #1142

## Follow-up Items

- Add another page `/guild/[guildname]/edit` rather than just linking to the `/join/guild` page. It works, but the UI is not totally accurate

### CODE REVIEW

File: hasura/migrations/1645304381138_move_guild_dao_addresses/up.sql
Comment (dan13ram) [inline hasura/migrations/1645304381138_move_guild_dao_addresses/up.sql:4]: We should ideally be calling this `gnosis-chain` or `gc` instead of `xdai`. But this is a change we have to do in multiple places, so perhaps best tracked in a separate PR. 

https://github.com/MetaFam/TheGame/issues/1170

Comment (dan13ram) [reply hasura/migrations/1645304381138_move_guild_dao_addresses/up.sql:4]: Let's also remember to double check these addresses are the only gnosis chain daos that we have onboarded so far while all of the rest are on mainnet, before we merged this to master.

File: packages/web/components/Guild/GuildHero.tsx
Comment (dan13ram) [inline packages/web/components/Guild/GuildHero.tsx:17]: @dysbulic on his PR has changed all occurences of `canEdit` with `editing`. So maybe good to rename this as well.

File: packages/web/pages/guild/[guildname].tsx
Comment (dan13ram) [inline packages/web/pages/guild/[guildname].tsx:50]: perhaps we can repace this useEffect and `canEdit` useState with a single useMemo hook.

File: packages/web/graphql/queries/guild.ts
Comment (dan13ram) [inline packages/web/graphql/queries/guild.ts:79]: Not used anywhere?

File: packages/web/components/Player/PlayerGuild.tsx
Comment (dan13ram) [inline packages/web/components/Player/PlayerGuild.tsx:40]: imo it would be cleaner to send linkProps as part of the props itself. 
```
type DaoHausLinkProps = { daoUrl: string | null } & LinkProps;
```
and we can destrucute the remaining props as already done below.

File: packages/web/components/Guild/GuildLinks.tsx
Comment (dan13ram) [inline packages/web/components/Guild/GuildLinks.tsx:18]: Are all guilds by default DaoHaus DAOs ?

File: packages/web/components/Guild/GuildForm.tsx
Comment (dan13ram) [inline packages/web/components/Guild/GuildForm.tsx:370]: We may want to make dao URL mandatory if this is not a DaoHaus dao. Perhaps a checkbox for the user to select if guild is DaoHaus dao or not?

**Review by dan13ram — COMMENTED** [review]
Most everything looks good to me. Lets wait for #1127 to be merged first though.

File: hasura/migrations/1645304381138_move_guild_dao_addresses/up.sql
Comment (alalonde) [reply hasura/migrations/1645304381138_move_guild_dao_addresses/up.sql:4]: > Let's also remember to double check these addresses are the only gnosis chain daos that we have onboarded so far while all of the rest are on mainnet, before we merged this to master.

I believe I pulled this data from the prod env, but will double check.

File: packages/web/graphql/queries/guild.ts
Comment (alalonde) [reply packages/web/graphql/queries/guild.ts:79]: removed

File: packages/web/pages/guild/[guildname].tsx
Comment (alalonde) [reply packages/web/pages/guild/[guildname].tsx:50]: good call

File: packages/web/components/Player/PlayerGuild.tsx
Comment (alalonde) [reply packages/web/components/Player/PlayerGuild.tsx:38]: Taken care of during the rebase

File: packages/web/components/Guild/GuildLinks.tsx
Comment (alalonde) [reply packages/web/components/Guild/GuildLinks.tsx:18]: No, but they can be associated with one or more

**Review by dan13ram — APPROVED** [review]

**Pull request merged**
